src/archetype.rs

```diff
@@ -43,11 +43,21 @@ pub struct Archetype {
 }
 
 impl Archetype {
-    pub(crate) fn new(types: Vec<TypeInfo>) -> Self {
+    pub(crate) fn empty() -> Self {
+        Self::new_(Vec::new())
+    }
+
+    pub(crate) fn new<B: crate::DynamicBundle>(types: Vec<TypeInfo>) -> Self {
         debug_assert!(
             types.windows(2).all(|x| x[0] < x[1]),
-            "type info unsorted or contains duplicates"
+            "attempted to allocate an entity with duplicate components ({})",
```

Comment:
This assert is conflating two separate conditions: misordering, which is a hecs bug, and duplicate components, which are a user bug. They should be separate asserts, with separately worded errors.

Reply:
Also, the equality check probably needs to be a non-debug `assert` for soundness.